code/modules/admin/verbs/freeze.dm

```diff
@@ -5,121 +5,95 @@
 //////Allows admin's to right click on any mob/mech and freeze them in place.///
 ////////////////////////////////////////////////////////////////////////////////
 ////////////////////////////////////////////////////////////////////////////////
-var/global/list/frozen_mob_list = list()
-/client/proc/freeze(var/mob/living/M as mob in GLOB.mob_list)
+var/global/list/frozen_atom_list = list()
+/client/proc/freeze(atom/movable/M)
 	set name = "Freeze"
 	set category = null
 
 	if(!check_rights(R_ADMIN))
 		return
 
-	if(!istype(M))
-		return
-
-	if(M in frozen_mob_list)
-		M.admin_unFreeze(src)
+	if(istype(M, /mob/living))
```

Comment:
perhaps instead of making it like this (harder to maintain) you can define admin_Freeze on movable level and make it return FALSE by default.
Then have living and mecha override it to actually do something and make it return TRUE (pods might also be a logical next step)
If it returns false you can display the message saying that it's not suited for this thing.

Reply:
Holy shit, I understand what you're saying now, and it works.

![image](https://user-images.githubusercontent.com/42044220/71320379-0b22f980-2470-11ea-9ca7-269790e8029a.png)